cmd/clusterctl/client/repository/components.go

```diff
@@ -199,6 +199,14 @@ func NewComponents(provider config.Provider, version string, rawyaml []byte, con
 		return nil, errors.Wrap(err, "failed to parse yaml")
 	}
 
+	// apply image overrides, if defined
+	objs, err = util.FixImages(objs, func(image string) (string, error) {
```

Comment:
For deployments that have multiple images, such as `capa-controller-manager`:
- gcr.io/kubebuilder/kube-rbac-proxy:v0.4.1
- us.gcr.io/k8s-artifacts-prod/cluster-api-aws/cluster-api-aws-controller:v0.5.0-rc.1

I think `FixImages()` would potentially set the same tag for both images, if you specified a tag in the image meta in the config?

Reply:
@ncdc the PR is not designed to prevent for user misconfigurations, but for making it really easier to address the main use case to change the image repository for all the images.

Then, if the need arises, we can introduce fine-grained configuration at image level of more sophisticated config validation

Reply:
That's fair. I just wanted to confirm this is the behavior. Let's make sure the image config section is very well documented. Basically users can't set the tag on anything other than cert-manager; otherwise, they'll mess up their cluster. Right?

Reply:
yes, that's right (assuming all the providers have more than 1 cimage)
I'm starting to work on a PR for documenting air-gapped support/the config, II will take care of documenting this properly

Reply:
xref https://github.com/kubernetes-sigs/cluster-api/pull/2595